tests/migrations/test_operations.py

```diff
@@ -1855,6 +1855,18 @@ def test_add_constraint(self):
         self.assertEqual(definition[1], [])
         self.assertEqual(definition[2], {'model_name': "Pony", 'constraint': gt_constraint})
 
+    @skipUnlessDBFeature('supports_table_check_constraints')
+    def test_add_or_constraint(self):
+        project_state = self.set_up_test_model("test_addorconstraint", index=True)
+        complex_check = models.Q(pink__gt=2, id__gt=0) | models.Q(id__gt=0)
+        complex_constraint = models.CheckConstraint(check=complex_check, name="test_add_constraint_nq")
+        complex_operation = migrations.AddConstraint("Pony", complex_constraint)
+        new_state = project_state.clone()
+        complex_operation.state_forwards("test_addorconstraint", new_state)
+        with connection.schema_editor() as editor:
+            complex_operation.database_forwards("test_addorconstraint", editor, project_state, new_state)
+        self.assertEqual(len(new_state.models["test_addorconstraint", "pony"].options["constraints"]), 1)
```

Comment:
You want to assert against the database level constraints here and not the `state` alterations. You can either use `connection.introspection.get_constraints` or trigger a constraint violation withing an `atomic` block like in `test_add_constraint`.

Reply:
`connection.introspection.get_constraints` assertion is more logical than before. I used it.